### PR TITLE
Automatically renew TLS certificates

### PR DESCRIPTION
**Pull request by super-cooper**
This change adds a new certbot Docker service which automatically attempts to renew the certs every day at 10 AM. nginx is reloaded every day by a cronjob on the host system. The only way to reload nginx is to send `SIGHUP` to its process, so certbot is unfortunately unable to do this without access to the Docker socket.

Certbot uses webroot challenges, and as such, nginx is configured to properly respond to Let's Encrypt's challenge requests.

### CODE REVIEW

**Comment by super-cooper** [comment]
[Automate Portal TLS certificate renewal](https://trello.com/c/UjaGiLvr/2790-automate-portal-tls-certificate-renewal)

File: conf/certbot/certbot.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM certbot/certbot:latest
+
+COPY crontab /etc/cron.d/certbot
+# Load the cronjob and create the log file to be able to run tail
+RUN crontab /etc/cron.d/certbot && touch /var/log/cron.log
+
+# Run cron, and keep an eye on the logs
+ENTRYPOINT ["crond", "&&", "tail", "-f", "/var/log/cron.log"]
```
Comment (msherman64) [inline conf/certbot/certbot.Dockerfile]: does `&&` work in an entrypoint?

Cron can be run in foreground with `cron -f`

**Comment by msherman64** [comment]
If we need a cronjob on the host system anyway, it's probably best to combine it with the certbot one.
Ideally again we only reload nginx when the cert is renewed, instead of once/twice per day.

Especially given that `/etc/letsencrypt` and `/var/www/certbot` are mounted in the nginx container, can we just run certbot on the host with `--deploy-hook docker restart nginx`?

**Review by msherman64 — CHANGES_REQUESTED** [review]
As in the comments, if we need a recurring task on the host anyway, it seems best to leverage certbot's deploy hook to only restart nginx on an actual change to the cert.

File: conf/certbot/certbot.Dockerfile (hunk 1 above)
Comment (super-cooper) [reply conf/certbot/certbot.Dockerfile]: It does work. Running cron in the foreground was not producing logs, so I switched to this method. I will look into putting this on the host instead.

**Comment by super-cooper** [comment]
The cronjob from the certbot image has been moved to host cron, and the renewal hook has been put into `/etc/letsencrypt/renewal-hooks/post/reload-nginx.sh`.

**Review by msherman64 — APPROVED** [review]
Looks good to me! Can you add the extra info about certbot to the Readme?
Otherwise, looks good to merge.

**Comment by super-cooper** [comment]
> Can you add the extra info about certbot to the Readme?

Done

**Pull request merged**